orion/sites/admin/src/programs/program.js

```diff
@@ -21,35 +21,93 @@ export class ProgramPage extends React.Component {
         });
     }
 
+    onClickSelectAll() {
+        var items = document.getElementsByName("unpublished");
+        for (var i = 0; i < items.length; i++) {
+            if (items[i].type == "checkbox") {
+                items[i].checked = true;
+            }
+        }
+    }
+
+    onClickPublish() {
+        console.log("clicked publish");
+    }
+
     render() {
         const rows = this.state.list.map((row, index) => {
             return <ProgramRow key={index} row={row} />;
         });
         const numRows = rows.length;
+        let numUnpublished = 0;
+        let numSelected = 0;
         return (
             <div id="view-program">
-                <h1>All Programs ({numRows})</h1>
+                <ul>
+                    <h1>All Programs ({numRows}) </h1>
+                    <p>
+                        {" "}
+                        You have {numUnpublished} unpublished items. You have
+                        selected {numSelected} items to publish.{" "}
+                    </p>
+                </ul>
                 <ul id="list-heading">
+                    <button
+                        className="li-small"
+                        onClick={this.onClickSelectAll}>
+                        Select All
+                    </button>
                     <li className="li-med">ProgramKey</li>
                     <li className="li-med">Name</li>
                     <li className="li-small">Grade1</li>
                     <li className="li-small">Grade2</li>
                 </ul>
                 <ul id="list-rows">{rows}</ul>
-                <Link className="add-program" to={"/programs/add"}>
-                    Add Program
-                </Link>
+                <ul id="list-buttons">
+                    <div className="li-med">
+                        <button>
+                            <Link className="add-program" to={"/programs/add"}>
+                                Add Program
+                            </Link>
+                        </button>
+                    </div>
+                    <div className="li-med">
+                        <button
```

Comment:
oh I see what you're trying to do. Ok so get rid of these `<div className="li-med">` stuff.

now give this button the CSS property `position: absolute` while the parent `<div id="buttons-container"/>` gets property `position: relative`.
From here, you can give the button `right: 0` and it will be the "right-most" area of its parent <- who in this case is "buttons-container". This also goes for top, left, and bottom. `top: 0, right: 0` means the upper right most area of the parent.

https://www.w3schools.com/css/css_positioning.asp